### PR TITLE
[NZCSA-023] Create-admin-route-and-api

### PR DESCRIPTION
**Pull request by Meloyg**
Issue: The event can't be modified.

Solution: Add modify event function for admin route

Risk: None

Review by: Garfield, Tony, Lucas

### CODE REVIEW

File: controllers/events.js [hunk 1]
```diff
@@ -34,14 +34,13 @@ exports.signUpRSVP = async (req, res, next) => {
     const lst = event.userList;
 
     // eslint-disable-next-line no-underscore-dangle
```
Comment (lucas2005gao) [inline controllers/events.js:36]: For these rules, if you don't want them you can chuck them into the ESLint config file. I can help with that later haha

File: controllers/events.js [hunk 2]
```diff
@@ -102,6 +101,44 @@ exports.addEvents = async (req, res, next) => {
...
+    const { isAdmin } = req.user;
+
+    if (!isAdmin) {
+      return next(new ErrorResponse('You are not admin', 401));
+    }
+
+    const selectedEvent = await Event.findById(eventId);
+
+    if (!selectedEvent) {
+      return next(new ErrorResponse('Event does not exist', 401));
```
Comment (lucas2005gao) [inline controllers/events.js]: the response code here should be 404, for item not found

**Review by lucas2005gao — CHANGES_REQUESTED** [review]
minor changes, rest looks good

File: controllers/events.js (hunk 1 above)
Comment (Meloyg) [reply controllers/events.js:36]: Haha sure

**Comment by lucas2005gao** [comment]
@ygua699 Make sure to get 2 approvals before merging pull requests!!

**Pull request merged**